genesis/streaming.py

```diff
@@ -14,350 +14,414 @@
 
 # FIXME: Make this into a proper class (safety in the unlikely case the user returns HEARTBEAT_SENTINEL)
 HEARTBEAT_SENTINEL = "__heartbeat__"
+
+
 def is_heartbeat(msg):
-	return isinstance(msg, str) and msg == HEARTBEAT_SENTINEL
+    return isinstance(msg, str) and msg == HEARTBEAT_SENTINEL
+
 
 assert is_heartbeat(HEARTBEAT_SENTINEL)
 
+
 def _noop(msg, meta):
-	return msg
+    return msg
+
 
 ## Message parsing
 
 Metadata = namedtuple("Metadata", "topic partition offset timestamp key idx")
 
+
 def parse_avro(val):
-	with io.BytesIO(val) as fp:
-		rr = fastavro.reader(fp)
-		for record in rr:
-			yield rr
+    with io.BytesIO(val) as fp:
+        rr = fastavro.reader(fp)
+        for record in rr:
+            yield rr
+
 
 def parse_json(val):
-	yield json.loads(val)
+    yield json.loads(val)
+
 
 def parse_blob(val):
-	yield val
+    yield val
+
 
-_MESSAGE_PARSERS = {
-	'avro': parse_avro,
-	'json': parse_json,
-	'blob': parse_blob
-}
+_MESSAGE_PARSERS = {"avro": parse_avro, "json": parse_json, "blob": parse_blob}
```

Comment:
This looks less readable than the original -- I'd keep the old version, as it's visually clear how many mappings there are, and the diff is cleaner as new ones are added.